resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1

```diff
@@ -231,6 +238,93 @@ class MousePointer {
     }
 }
 
+[DSCResource()]
+class AnimationEffects {
+	#Need to verify the Animation Effects setting toggle on the Accessibility page changes when these are updated, or find the setting to update that also.
+    [DscProperty(Key)] [AnimationEffectsState] $AnimationEffectsState = [AnimationEffectsState]::KeepCurrentValue
+    [DscProperty(NotConfigurable)] [int] $SmoothScrollListBoxes
+    [DscProperty(NotConfigurable)] [int] $SlideOpenComboBoxes
+    [DscProperty(NotConfigurable)] [int] $FadeOrSlideMenusIntoView
+    [DscProperty(NotConfigurable)] [int] $ShowShadowsUnderMousePointer
+    [DscProperty(NotConfigurable)] [int] $FadeOrSlideToolTipsIntoView
+    [DscProperty(NotConfigurable)] [int] $FadeOutMenuItemsAfterClicking
+    [DscProperty(NotConfigurable)] [int] $ShowShadowsUnderWindows
+
+    hidden [string] $AnimationEffectsProperty = 'UserPreferencesMask'
+	#These settings are stored alongside other settings in a bitmask.
+
+    [AnimationEffects] Get() {
+        $currentState = [AnimationEffectsState]::new()
+
+		$AnimationState = (Get-ItemPropertyValue -Path $global:AnimationEffectsSettingsRegistryPath -Name $AnimationEffectsProperty) | %{[System.Convert]::ToString($_,2).PadLeft(8,'0')} #Get-ItemPropertyValue converts hex to int, so need to complete converting to binary.
+
+		#Decode bitmask string array as char array grid
+		#1001ABC0 
+		#00D1EF10 
+		#00000G11
+
+		$SmoothScrollListBoxes = $AnimationState[0][4] #A
+		$SlideOpenComboBoxes = $AnimationState[0][5] #B
+		$FadeOrSlideMenusIntoView = $AnimationState[0][6] #C
+
+		$ShowShadowsUnderMousePointer = $AnimationState[1][2] #D
+		$FadeOrSlideToolTipsIntoView = $AnimationState[1][4] #E
+		$FadeOutMenuItemsAfterClicking = $AnimationState[1][5] #F
+
+		$ShowShadowsUnderWindows = $AnimationState[2][5] #G
+
+		if ($ShowShadowsUnderWindows -eq 0 
+		-AND $SlideOpenComboBoxes -eq 0 
+		-AND $FadeOrSlideMenusIntoView -eq 0 
+		-AND $ShowShadowsUnderMousePointer -eq 0 
+		-AND $FadeOrSlideToolTipsIntoView -eq 0 
+		-AND $FadeOutMenuItemsAfterClicking -eq 0 
+		-AND $ShowShadowsUnderWindows -eq 0) {
+			$currentState = [AnimationEffectsState]::Disabled
+		} else {
+			$currentState = [AnimationEffectsState]::Enabled
+		}
+
+		return $currentState
+	}
+
+    [bool] Test() {
+		$currentState = $this.Get()
+		if ($this.AnimationEffectsState -ne [AnimationEffectsState]::KeepCurrentValue -and $this.AnimationEffectsState -ne $currentState.AnimationEffectsState) {
+			return $false
+		}
+
+		return $true
+    }
+
+    [void] Set() {
+		if ($this.AnimationEffectsState -ne [AnimationEffectsState]::KeepCurrentValue) {
+            $desiredValue = switch ([AnimationEffectsState]($this.AnimationEffectsState)) {
+                Enabled {1}
+                Disabled {0}
+            }

```

Comment:
You should be able to call the Test method directly here and check the boolean result to see if we are in a desired state.

Reply:
I have added this check.